suse_migration_services/migration_config.py

```diff
@@ -43,8 +46,32 @@ def __init__(self):
             Defaults.get_migration_config_file()
         self.migration_custom_file = \
             Defaults.get_system_migration_custom_config_file()
-        with open(self.migration_config_file, 'r') as config:
-            self.config_data = yaml.safe_load(config)
+        self.config_data = self._parse_config_file(self.migration_config_file)
+
+    def _parse_config_file(self, config_file):
+        config_data = {}
+        with open(config_file, 'r') as config:
+            try:
+                validator = Validator(schema)
+                config_data = yaml.safe_load(config)
+                if config_data is None:
+                    # Config file is empty (or all comments), reset to empty dict
+                    config_data = {}
+                validator.validate(config_data)
+            except Exception as e:
+                message = 'Loading {0} failed: {1}: {2}'.format(
+                    config_file, type(e).__name__, e
+                )
+                log.error(message)
+                raise DistMigrationConfigDataException(message)
+            validator.validate(config_data)
```

Comment:
A second validation of the same config data. Given we have the same validator object than we had inside the `try:` block this second validation is superfluous.

Reply:
Oops, copy-paste error while I was moving things around :-)